homeassistant/components/mqtt/discovery.py

```diff
@@ -311,10 +311,7 @@ async def async_integration_message_received(
                     and result["reason"]
                     in ("already_configured", "single_instance_allowed")
                 ):
-                    unsub = mqtt_data.integration_unsubscribe.pop(key, None)
-                    if unsub is None:
-                        return
-                    unsub()
+                    mqtt_data.integration_unsubscribe.pop(key)()
```

Comment:
There is already a check for `key` not being available in `mqtt_data.integration_unsubscribe`, so unsub cannot be `None` here.

https://github.com/home-assistant/core/pull/91796/files#diff-189ecd07b98c2a9602a647f97dfe63e919c95af2d3856701044958b230591ef1R294-R295